feat(ColorMaps): Add grayscale sigmoid colormap based on DICOM standard

jadh4v opened the pull request:
<!--
👋 Hello, and thank you for starting this contribution!
📖 Make sure you've read our [CONTRIBUTING.md](https://github.com/Kitware/vtk-js/blob/master/CONTRIBUTING.md) guide before submitting your pull request.
❗️ Please follow the template below to help other contributors review your work.
-->

# Context
<!--
Explain why this change is needed. Please include relevant links supporting this change, such as:
- fix #ISSUE_NUMBER (from issue tracker)
- discourse post thread, or any other existing references
- screenshot of the issue
- console log of error, callstack
-->

Add a non-linear sigmoid grayscale colormap as an alternative to linear grayscale. This mapping is described in the DICOM standard in section C.11.2.1.3.1 SIGMOID Function and equation C.11-1. We use a normalized window_width of [-1, 1] and window_center = 0. The colormap is generated using a 16-point sampling of this function.

See: https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_c.11.2.html

~~## There are two ways to access the sigmoid function:~~
~~1. Developer can use the newly added sigmoid presets that already are generated on the sigmoid curve, or
2. Use the `ScalarMappingMode` property on `vtkColorTransferFunction` to adapt any color map to sigmoid mapping. the `vtkColorTransferFunction` provides two `ScalarMappingModes`. `LINEAR` continues to be the default value and behavior, and `SIGMOID` allows to switch the scalar to color mapping to the sigmoid function. Additionally, you can control the growth rate factor using `setSigmoidGrowthRate()` which is usually a negative number and equal to `-4` according to the DICOM standard.~~

# Results
<!--
Describe or illustrate the effects of your contribution. Please include:
- comparisons of the behavior before vs after
- screenshots of new or changed visualizations if applicable
-->
A comparison of linear vs sigmoid grayscale using the ImageViewer example.
```
npm run example ImageViewer
```

| linear | sigmoid |
|--|--|
| <img width="400" alt="Screenshot 2025-11-18 at 6 49 43 PM" src="https://github.com/user-attachments/assets/54643588-0f89-44f8-ad3a-cc9703423c74" /> | <img width="400" alt="Screenshot 2025-11-18 at 6 50 00 PM" src="https://github.com/user-attachments/assets/1d8a8758-2cdc-4f9f-b616-a4674e923abe" /> |

The sampled 16-point sigmoid function plotted through matplotlib:
<img width="640" height="480" alt="dicom-grayscale-sigmoid-plot" src="https://github.com/user-attachments/assets/1ace050c-2d46-462a-9be6-74cca004654a" />


### Changes
<!--
Please describe what is changing. Include:
- APIs added, deleted, deprecated, or changed
- Classes and methods added, deleted, deprecated, or changed
- A summary of usage if this is a new feature or change to an API. Adequate documentation and TS definitions should also be added/updated.
-->
- [x] Documentation and TypeScript definitions were updated to match those changes

### PR and Code Checklist
<!--
NOTE: We will not merge if the following steps have not been completed!
-->
- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Testing
<!--
Please describe how this can be tested by reviewers. Be specific about anything not tested and the reasons why.
Tests should complete without errors. See CONTRIBUTING.md
-->
- [ ] This change adds or fixes unit tests <!-- Tests should be added for new functionality -->
- [ ] Tested environment:
  - **vtk.js**: <!-- ex: 14.0.0 (favor latest master) -->
  - **OS**: <!-- ex: Windows 10, iOS 13.6 -->
  - **Browser**: <!-- ex: Chrome 89.0.4389.128 -->

<!--
Edit and uncomment the section below if relevant

### Funding
This contribution is funded by [Example](https://example.com).

 -->

finetjul commented on Sources/Rendering/Core/ColorTransferFunction/ColorMaps.json:
```diff
@@ -529,6 +529,78 @@
+         1,
+         0.9820137900379085,
+         0.9820137900379085,
+         0.9820137900379085
```
why not 1 ?

finetjul commented on Sources/Rendering/Core/ColorTransferFunction/ColorMaps.json:
```diff
@@ -529,6 +529,78 @@
       "RGBPoints" : [ 0, 0, 0, 0, 1, 1, 1, 1 ]
    },
+   {
+      "ColorSpace" : "RGB",
```
Maybe add support for a "Nodes" key/value in addition to "RGBPoints"

jadh4v commented on Sources/Rendering/Core/ColorTransferFunction/ColorMaps.json:
```diff
@@ -529,6 +529,78 @@
+         -1,
+         0.01798620996209156,
+         0.01798620996209156,
+         0.01798620996209156,
```
I think we could rescale the function to map from 0 to 1 so that it uses the full available grayscale range, but I am not certain if that is ok to do for the DICOM standard. I think we could wait to fix this once we get more feedback on this feature from developers.
Or, I have already added `LINEAR` and `SIGMOID` modes to `vtkColorTransferFunction`, I can add another `RESCALED_SIGMOID`.

jadh4v commented:
@finetjul 
I have updated this PR to drop the TransferFunction related changes. Sigmoid is now only available through the presets. Added equivalent `inverted-grayscale` and `inverted-grayscale-DICOM-sigmoid` presets. I also rebased the branch on the latest main. Should be ready to merge when ci pipeline passes.

finetjul reviewed (approved):
LGTM
